lib/spack/docs/basic_usage.rst

```diff
@@ -1226,17 +1229,27 @@ variants using the backwards compatibility syntax and uses only ``~``
 for disabled boolean variants.  The ``-`` and spaces on the command
 line are provided for convenience and legibility.
 
+Spack allows variants to propagate their value to the package's
+dependency by using ``++``, ``--``, and ``~~`` for boolean variants,
+and ``name==value`` for non-boolean variants. For example:
+
+.. code-block:: sh
+
+    mpileaks ++debug   # enabled debug will be propagated to dependecies
+    mpileaks +debug    # only mpileaks will have debug enabled
+
 ^^^^^^^^^^^^^^
 Compiler Flags
 ^^^^^^^^^^^^^^
 
 Compiler flags are specified using the same syntax as non-boolean variants,
 but fulfill a different purpose. While the function of a variant is set by
 the package, compiler flags are used by the compiler wrappers to inject
-flags into the compile line of the build. Additionally, compiler flags are
-inherited by dependencies. ``spack install libdwarf cppflags="-g"`` will
-install both libdwarf and libelf with the ``-g`` flag injected into their
-compile line.
+flags into the compile line of the build. Additionally, compiler flags can
+be inherited by dependencies by using ``==``. Though, versions of spack prior to
+0.19.0 will use the ``=`` syntax.
```

Comment:
Perhaps the reference to pre-0.19.0 versions could be added as a `note` for emphasis instead?

Reply:
@tldahlgren I'm confused on what a reference to pre-0.19.0 would be. Do you mean a link, or something else?

Reply:
Resolving since we discussed this off-line.